src/dev/fsys.c

```diff
@@ -332,7 +332,7 @@ short fsys_findnext(short dir, p_file_info file) {
     FRESULT fres;
 
     if (g_dir_state[dir]) {
-        fres = f_findnext(&g_dir_state[dir], &finfo);
+        fres = f_findnext(&g_directory[dir], &finfo);
```

Comment:
This seems like a change in functionality, The variable name is different. Did you confirm this works?

Reply:
No I didn't test because I didn't find a code path (but I didn't try too hard) that would call this function, but I think it could not work before. g_dir_state is an array of chars while f_findnext (from FFS) expects a DIR pointer.